### PR TITLE
Adds Railtie helper 

### LINKED ISSUE #66: Automatically load Railtie when gem is included in Rails
[Marginalia has an example](https://github.com/basecamp/marginalia/blob/master/lib/marginalia/railtie.rb) how to automatically load a Railtie when Rails is present. This makes configuring the gem a lot easier.

Maybe we can use the [appraisal gem](https://github.com/thoughtbot/appraisal) to test with and without Rails.

### PR DESCRIPTION
**Pull request by Willianvdv**
Fixes #66

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,3 +7,5 @@ end
 
 # Specify your gem's dependencies in arel_toolkit.gemspec
 gemspec
+
+gem 'rb-readline'
```
Comment (mvgijssel) [inline Gemfile]:
```suggestion
```

File: lib/arel/middleware/railtie.rb [hunk 2]
```diff
@@ -1,4 +1,16 @@
 module Arel
+  if defined? Rails::Railtie
```
Comment (mvgijssel) [inline lib/arel/middleware/railtie.rb]: Looking forward to some tests around this! Dummy app? Rails Rspec? 

You can add the Rails gem to the appraisal `arel_gems` group, so it's testing specs with and without Rails being loaded.

File: spec/railtie_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  describe 'index' do
+    it 'returns a sql generated by the arel toolkit' do
+      User.create! username: 'Maarten'
+      User.create! username: 'Willian'
+
+      puts User.all.map &:username
+
+      get :index
+
+      expect(response.body).to eq ['Willian'].to_json
```
Comment (Willianvdv) [inline spec/railtie_spec.rb:33]: All this work for only this assertion :)

File: arel_toolkit.gemspec [hunk 4]
```diff
@@ -44,7 +45,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rubocop', '= 0.71.0'
   spec.add_development_dependency 'guard', '~> 2.15'
   spec.add_development_dependency 'guard-rspec', '~> 4.7'
-  spec.add_development_dependency 'guard-rubocop', '~> 1.3.0'
```
Comment (mvgijssel) [inline arel_toolkit.gemspec]: Don't like this gem?

File: spec/railtie_spec.rb (hunk 3 above)
Comment (mvgijssel) [reply spec/railtie_spec.rb:33]: 😍

File: spec/dummy/app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+class Post < ActiveRecord::Base
```
Comment (Willianvdv) [inline spec/dummy/app/models/post.rb]: should inherit from application record

File: spec/dummy/config/environment.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Load the Rails application.
+require_relative 'application'
+
+load File.expand_path('../../../lib/arel_toolkit.rb', __dir__)
```
Comment (Willianvdv) [inline spec/dummy/config/environment.rb:4]: Since this file is already required by the spec, instead of requiring, I'm (re)loading the arel_toolkit here to allow railtie to load as Rails is now available.

File: spec/dummy/app/models/application_record.rb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+class ApplicationRecord < ActiveRecord::Base
+  self.table_name_prefix = 'dummy_'
```
Comment (Willianvdv) [inline spec/dummy/app/models/application_record.rb:2]: @mvgijssel the dummy's models are prefixed to prevent name collisions with models defined in the specs

File: arel_toolkit.gemspec (hunk 4 above)
Comment (Willianvdv) [reply arel_toolkit.gemspec]: get it out of here

File: lib/arel_toolkit/railtie.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+module ArelToolkit
+  class Railtie < ::Rails::Railtie
+    initializer 'ArelToolkit.insert' do
+      ActiveSupport.on_load :active_record do
+        Arel::Middleware::Railtie.insert_postgresql
```
Comment (Willianvdv) [inline lib/arel_toolkit/railtie.rb:5]: @mvgijssel why does this middleware lives in `Arel` and not in `ArelToolkit`?

File: spec/pg_search_spec.rb [hunk 9]
```diff
@@ -89,9 +89,9 @@ def self.call(arel, _context)
 
     it 'works for searching associations' do
       Arel.middleware.apply([SomeMiddleware]) do
-        salami1 = Salami.create!
-        salami2 = Salami.create!
-        salami3 = Salami.create!
+        salami1 = Salami.create! cracker: Cracker.create!
```
Comment (Willianvdv) [inline spec/pg_search_spec.rb:92]: I'm not sure why, but Salami now needs a Cracker.

File: lib/arel_toolkit/railtie.rb (hunk 8 above)
Comment (mvgijssel) [reply lib/arel_toolkit/railtie.rb:5]: Everything is an addition to the Arel namespace, the ArelToolkit namespace is there for naming purposes of the gem.

File: spec/pg_search_spec.rb (hunk 9 above)
Comment (mvgijssel) [reply spec/pg_search_spec.rb:92]: 🤣

File: spec/spec_helper.rb [hunk 10]
```diff
@@ -18,6 +18,15 @@
 require 'support/pg_ast_contains'
 require 'support/visitors'
 
+require File.expand_path('../dummy/config/environment', __FILE__)
```
Comment (mvgijssel) [inline spec/spec_helper.rb:21]: Why is this necessary (till line 29)

**Review by mvgijssel — COMMENTED** [review]
Would it be possible to move `spec/dummy` to `spec/support/dummy` 😏?

File: .travis.yml [hunk 11]
```diff
@@ -21,6 +21,7 @@ install:
   - GEM_HOME=vendor/bundle/ruby/2.5.0 bundle install --jobs=3 --retry=3 --deployment --path=${BUNDLE_PATH:-vendor/bundle}
   - GEM_HOME=vendor/bundle/ruby/2.5.0 bundle exec appraisal install
 script:
+  - GEM_HOME=vendor/bundle/ruby/2.5.0 bundle exec rails db:setup
```
Comment (mvgijssel) [inline .travis.yml:24]: As you maintain the test schema below, why is this necessary?

File: arel_toolkit.gemspec [hunk 12]
```diff
@@ -34,6 +34,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'github_changelog_generator', '~> 1.14.3'
   spec.add_development_dependency 'rake', '~> 10.0'
   spec.add_development_dependency 'rspec', '~> 3.8'
+  spec.add_development_dependency 'rspec-rails', '~> 3.8'
```
Comment (mvgijssel) [inline arel_toolkit.gemspec:37]: So adding this gem will cause Rails to be included as well?

**Review by mvgijssel — APPROVED** [review]

File: arel_toolkit.gemspec (hunk 12 above)
Comment (Willianvdv) [reply arel_toolkit.gemspec:37]: Looking at the lock file, it includes 
```
      actionpack (>= 3.0)
      activesupport (>= 3.0)
      railties (>= 3.0)
```
No Rails I would think

File: .travis.yml (hunk 11 above)
Comment (Willianvdv) [reply .travis.yml:24]: I think to create the database. Without this line, the build fails but I've not validated whether the maintain schema thing creates db-s or not.
